### PR TITLE
appservice OneDeploy

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
API review.

Not fully tested, but at least this works for tomcat.
```
            WebApp app = azure.webApps().getByResourceGroup("rg-weidxu", "wa1weidxu");
            app.deploy(DeployType.ZIP, new File(ManageWebAppBasic.class.getResource("/helloworld.zip").getPath()));
            app.deploy(DeployType.WAR, new File(ManageWebAppBasic.class.getResource("/helloworld.war").getPath()), new DeployOptions().withPath("webapps/app1"));
```

And this works for java SE.
```
            app.deploy(DeployType.JAR, new File(ManageWebAppBasic.class.getResource("/some_sprint_app.jar").getPath()));
```

PS: OneDeploy https://github.com/shrishrirang/onedeploy

### CODE REVIEW

File: azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/SupportsOneDeploy.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import rx.Completable;
+
+import java.io.File;
+import java.io.InputStream;
+
+/**
+ * Provides access to OneDeploy.
+ */
+@Beta(Beta.SinceVersion.V1_36_0)
+public interface SupportsOneDeploy {
```
Comment (weidongxu-microsoft) [inline azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/SupportsOneDeploy.java:19]: API here.

File: azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/DeployOptions.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+/**
+ * The options for OneDeploy.
+ */
+public class DeployOptions {
+
+    private String path;
+    private Boolean restartSite;
+    private Boolean cleanDeployment;
+    private Boolean ignoreStack;
```
Comment (ChenTanyi) [inline azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/DeployOptions.java:17]: Is it used?

**Review by ChenTanyi — COMMENTED** [review]
LGTM. May need add some test case.

File: azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/DeployOptions.java (hunk 2 above)
Comment (weidongxu-microsoft) [reply azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/DeployOptions.java:17]: Oh, it is not. I will remove it.

**Comment by weidongxu-microsoft** [comment]
At present, not able to add live test, since web app is not ready (until about Nov.), manually script need to be called to prepare the site.

**Comment by weidongxu-microsoft** [comment]
DeploymentSlot
```
            slot.deploy(DeployType.JAR_LIB, new File("C:\\github_fork\\autorest.java\\preprocessor\\target\\preprocessor.jar"), new DeployOptions().withPath("preprocessor.jar"));
            slot.deploy(DeployType.STATIC, new File("C:\\github_fork\\autorest.java\\generate.bat"), new DeployOptions().withPath("generate.bat"));
            slot.deploy(DeployType.STATIC, new File("C:\\github_fork\\autorest.java\\generate.bat"), new DeployOptions().withPath("generate.bat"));
            slot.deploy(DeployType.SCRIPT_STARTUP, new File("C:\\github_fork\\autorest.java\\generate.bat"));
```

**Comment by weidongxu-microsoft** [comment]
Manual test is good.
@xccc-msft @ChenTanyi

**Review by ChenTanyi — APPROVED** [review]

**Pull request merged**